systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java

```diff
@@ -61,5 +63,11 @@ private static long timeoutForClientFinishJob(int messagesCount) {
         // need to add at least 1-2minutes for finishing the job
         return (long) messagesCount * 1000 + Duration.ofMinutes(2).toMillis();
     }
+
+    public static String generateRandomConsumerGroup() {
+        String salt = new Random().nextInt(Integer.MAX_VALUE) + "-" + new Random().nextInt(Integer.MAX_VALUE);
```

Comment:
I had the same discussion with @see-quick about using two "salts" somehow. I don't think it's useful.
The `Random` class implements a PRNG (Pseudorandom number generator) so it means that when you use a `nextXXX` method, it's going to generate the next pseudorandom number in the "sequence", so every `nextXXX` in the same execution will return you a different number. There is no way it returns the same number as before.
The same sequence could be generated in two different runs of the tests if the "seed" used for initializing the pseudorandom sequence generation is the same. It doesn't happen with the default `Random` constructor because internally it uses system time as part of the seed, so it's different for every run so the generated sequence of random numbers is different every run.
Anyway, I think that concatenating two random numbers is useless.

Another point is that you should define one `Random` instance in the class using it. Why creating a new instance every time `generateRandomConsumerGroup` is called? As I said, it provides you a sequence of random number so each call to `nextXXX` on the same `Random` instance cannot return an already generated random number.

Reply:
Thanks for this explanation, it makes more sense now for me.